src/sentry/integrations/discord/message_builder/issues.py

```diff
@@ -55,24 +57,48 @@ def build(self, notification_uuid: str | None = None) -> dict[str, object]:
         rule_id = None
         rule_environment_id = None
         if self.rules:
-            rule_id = self.rules[0].id
             rule_environment_id = self.rules[0].environment_id
+            if features.has("organizations:workflow-engine-ui-links", self.group.organization):
+                rule_id = int(get_key_from_rule_data(self.rules[0], "workflow_id"))
+            elif features.has(
+                "organizations:workflow-engine-trigger-actions", self.group.organization
+            ):
+                rule_id = int(get_key_from_rule_data(self.rules[0], "legacy_rule_id"))
+            else:
+                rule_id = self.rules[0].id
```

Comment:
I'm really starting to think we could use a utility method for this now. Seems like we implement this logic in a few places.